### PR TITLE
fix(SiriusXM): fix and rewrite for new web player

### PR DESCRIPTION
**Pull request by itzaidan**
## Description 

The SiriusXM presence has not been working since they updated their web player back in December so I've gone ahead and rewrote and fixed the presence as well as added album/channel artwork depending on what page the user is on or listening to, as well as an elapsed timer when the user starts listening, which stops when they have paused.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

msedge 05-21-2024 08-01-13
msedge 05-21-2024 08-03-27

</details>

### CODE REVIEW

File: websites/S/SiriusXM/metadata.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
 		{
 			"name": "Alexx",
 			"id": "604182077088202784"
+		},
+		{
+			"name": "Aidan",
+			"id": "235532241403510795"
```
Comment (DarkVillager) [inline websites/S/SiriusXM/metadata.json:14]: Feel free to takeover the presence if you want

File: websites/S/SiriusXM/metadata.json [hunk 2]
```diff
@@ -16,12 +20,12 @@
...
-		"player.siriusxm.com",
-		"player.siriusxm.ca"
+		"can.siriusxm.com",
+		"www.siriusxm.com"
 	],
-	"version": "1.2.15",
-	"logo": "https://cdn.rcd.gg/PreMiD/websites/S/SiriusXM/assets/logo.jpg",
-	"thumbnail": "https://cdn.rcd.gg/PreMiD/websites/S/SiriusXM/assets/thumbnail.png",
+	"version": "2.0.0",
+	"logo": "https://i.imgur.com/WVRaXaV.png",
```
Comment (theusaf) [inline websites/S/SiriusXM/metadata.json:27]: Logos should be 512x512px

File: websites/S/SiriusXM/presence.ts [hunk 3]
```diff
@@ -1,86 +1,116 @@
...
+	const presenceData: PresenceData = { largeImageKey: sxmLogo, startTimestamp },
+		title = getTextContent(
+			"#root > div > div._BaseLayout_711jo_11 > main > div > div > header > div > div > div._EntityHeaderMetadata_1mpio_27 > div._EntityHeaderTexts_1mpio_32 > div._EntityHeaderTitle_1mpio_43 > h1"
+		),
+		subtitle = getTextContent(
+			"#root > div > div._BaseLayout_711jo_11 > main > div > div > header > div > div > div._EntityHeaderMetadata_1mpio_27 > div._EntityHeaderTexts_1mpio_32 > div._EntityHeaderSubtitle_1mpio_47"
+		),
+		channelImage = getImageSrc(
+			"#root > div > div._BaseLayout_711jo_11 > main > div > div > header > div > div > div._EntityImageContainer_1mpio_65 > span > span > span > span"
+		),
```
Comment (theusaf) [inline websites/S/SiriusXM/presence.ts:41]: Ideally, use less specific selectors and avoid selectors including computer-generated class names. A useful tool is the [`*=`](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attrvalue_6) selector

File: websites/S/SiriusXM/presence.ts [hunk 4]
```diff
@@ -1,86 +1,116 @@
...
+		path = Object.keys(pathDetailsMap).find(p =>
+			document.location.pathname.includes(p)
+		);
+	if (path) {
+		presenceData.details = pathDetailsMap[path];
+		if (path === "/player/search") {
+			presenceData.state = (<HTMLInputElement>(
+				document.querySelector("._SearchBarInput_k2jza_6")
+			))?.value;
```
Comment (theusaf) [inline websites/S/SiriusXM/presence.ts:50]: Use generics rather than casting

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/S/SiriusXM/metadata.json (hunk 2 above)
Comment (itzaidan) [reply websites/S/SiriusXM/metadata.json:27]: Whoops! I didn't realize the image wasn't scaled to 512x when I uploaded it.

**Comment by theusaf** [comment]
Closing for inactivity. You may open a new PR after resolving the issues mentioned in this one.

**Pull request closed without merging**